dist/log.py

```diff
@@ -6,390 +6,652 @@
+static inline int
+__wt_struct_pack_%(name)s(uint8_t **pp, uint8_t *end%(comma)s
+    %(arg_decls_in)s)
+{
+    %(pack_body)s
+    return (0);
+}
+
+/*
+ * __wt_struct_unpack_%(name)s --
+ *     Unpack the %(name)s struct.
+ */
+WT_GCC_FUNC_DECL_ATTRIBUTE((warn_unused_result))
+static inline int
+__wt_struct_unpack_%(name)s(const uint8_t **pp, const uint8_t *end%(comma)s
+    %(arg_decls_out)s)
+{
+    %(unpack_body)s
+    return (0);
+}
+
+
 /*
  * __wt_logop_%(name)s_pack --
- *\tPack the log operation %(name)s.
+ *     Pack the log operation %(name)s.
  */
+WT_GCC_FUNC_DECL_ATTRIBUTE((warn_unused_result))
 int
 __wt_logop_%(name)s_pack(
     WT_SESSION_IMPL *session, WT_ITEM *logrec%(comma)s
-    %(arg_decls)s)
+    %(arg_decls_in)s)
 {
-\tconst char *fmt = WT_UNCHECKED_STRING(%(fmt)s);
-\tsize_t size;
-\tuint32_t optype, recsize;
-
-\toptype = %(macro)s;
-\tWT_RET(__wt_struct_size(session, &size, fmt,
-\t    optype, 0%(pack_args)s));
-
-\t__wt_struct_size_adjust(session, &size);
-\tWT_RET(__wt_buf_extend(session, logrec, logrec->size + size));
-\trecsize = (uint32_t)size;
-\tWT_RET(__wt_struct_pack(session,
-\t    (uint8_t *)logrec->data + logrec->size, size, fmt,
-\t    optype, recsize%(pack_args)s));
-
-\tlogrec->size += (uint32_t)size;
-\treturn (0);
+    size_t size;
+    uint8_t *buf, *end;
+
+    size = __wt_struct_size_%(name)s(%(pack_args)s);
+    size += __wt_vsize_uint(%(macro)s) + __wt_vsize_uint(0);
+    __wt_struct_size_adjust(session, &size);
+    WT_RET(__wt_buf_extend(session, logrec, logrec->size + size));
+
+    buf = (uint8_t *)logrec->data + logrec->size;
+    end = buf + size;
+    WT_RET(__wt_logop_write(session, &buf, end, %(macro)s, (uint32_t)size));
+    WT_RET(__wt_struct_pack_%(name)s(&buf, end%(comma)s%(pack_args)s));
+
+    logrec->size += (uint32_t)size;
+    return (0);
 }
-''' % {
-    'name' : optype.name,
-    'macro' : optype.macro_name(),
-    'comma' : ',' if optype.fields else '',
-    'arg_decls' : ', '.join(
-        '%s%s%s' % (cintype(f), '' if cintype(f)[-1] == '*' else ' ', f[1])
-        for f in optype.fields),
-    'pack_args' : ''.join(', %s' % pack_arg(f) for f in optype.fields),
-    'fmt' : op_pack_fmt(optype),
-})
 
-    tfile.write('''
 /*
  * __wt_logop_%(name)s_unpack --
- *\tUnpack the log operation %(name)s.
+ *     Unpack the log operation %(name)s.
  */
 int
 __wt_logop_%(name)s_unpack(
     WT_SESSION_IMPL *session, const uint8_t **pp, const uint8_t *end%(comma)s
-    %(arg_decls)s)
+    %(arg_decls_out)s)
 {
-\tWT_DECL_RET;
-\tconst char *fmt = WT_UNCHECKED_STRING(%(fmt)s);
-\tuint32_t optype, size;
-
-\tif ((ret = __wt_struct_unpack(session, *pp, WT_PTRDIFF(end, *pp), fmt,
-\t    &optype, &size%(unpack_args)s)) != 0)
-\t\tWT_RET_MSG(session, ret, "logop_%(name)s: unpack failure");
-\tWT_ASSERT(session, optype == %(macro)s);
-
-\t*pp += size;
-\treturn (0);
+    WT_DECL_RET;
+    uint32_t optype, size;
+
+#if !defined(NO_STRICT_PACKING_CHECK) || defined(PACKING_COMPATIBILITY_MODE)
```

Comment:
can you add a comment at the top of this file to explain the use of these macros? Also, could you mention that these will be removed in future versions? I do not find any need for these macros other than testing, but we can keep them for one major release and remove them.

Reply:
Done